File: sdk/android/src/main/java/com/dualler/sdk/handler/ToastHandler.kt
package com.dualler.sdk.handler

import android.app.Activity
import android.widget.Toast
import com.dualler.sdk.protocol.APIContract
import com.dualler.sdk.protocol.BridgeMessage

class ToastHandler(private val activity: Activity) : Handler {
    override val apiName: String = APIContract.SHOW_TOAST

    override fun handle(message: BridgeMessage): String? {
        val payload = message.payload as? Map<*, *> ?: return null
        val title = payload["title"] as? String ?: ""
        val duration = (payload["duration"] as? Number)?.toInt() ?: 2000
        val icon = payload["icon"] as? String ?: "none"

        val toastDuration = when (icon) {
            "success" -> Toast.LENGTH_SHORT
            "loading" -> Toast.LENGTH_LONG
            else -> Toast.LENGTH_SHORT
        }

        Toast.makeText(activity, title, toastDuration).show()
        return "{}"
    }
}

File: sdk/ios/Sources/DuallerSDK/ToastHandler.swift
import UIKit

/// Shows toast notifications to the user
public final class ToastHandler: Handler {
    public let apiName: String = "dualler.showToast"

    public func handle(_ message: BridgeMessage) -> String? {
        guard let payload = message.payload else { return "{}" }
        let title = payload["title"] as? String ?? ""
        let duration = (payload["duration"] as? TimeInterval) ?? 2.0

        DispatchQueue.main.async {
            let vc = UIApplication.shared.connectedScenes
                .compactMap { $0 as? UIWindowScene }
                .flatMap { $0.windows }
                .first { $0.isKeyWindow }

            let toast = UIAlertController(title: nil, message: title, preferredStyle: .alert)
            // Center-toast styling via positioning would go here in production
            vc?.present(toast, animated: true)
            DispatchQueue.main.asyncAfter(deadline: .now() + duration) {
                toast.dismiss(animated: false)
            }
        }

        return "{}"
    }
}
